### PR TITLE
sea-orm-cli: init at 0.11.3

### PR DESCRIPTION
**Pull request by traxys**
###### Description of changes

Sea-orm-cli is a tool to help write [SeaORM](https://www.sea-ql.org/SeaORM/) applications.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by traxys** [comment]
Result of `nixpkgs-review pr 231339` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>sea-orm-cli</li>
  </ul>
</details>

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 231339` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>sea-orm-cli</li>
  </ul>
</details>

**Comment by traxys** [comment]
I don't have a darwin machine to test on, but I suspect it is an issue with TLS. Do you have any logs, or should I mark the package linux only ?

File: pkgs/development/tools/sea-orm-cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  version = "0.11.3";
+
+  src = fetchCrate {
+    inherit pname version;
+    hash = "sha256-VRSdPsjRubJOsjdAxdnFCM9VmAVwGkXDvpXT4GF2jxY=";
+  };
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ openssl ];
```
Comment (wegank) [inline pkgs/development/tools/sea-orm-cli/default.nix:19]:
```suggestion
  buildInputs = [ openssl ]
    ++ lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.SystemConfiguration ];
```

File: pkgs/development/tools/sea-orm-cli/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{ lib
+, rustPlatform
+, fetchCrate
+, pkg-config
+, openssl
+,
```
Comment (wegank) [inline pkgs/development/tools/sea-orm-cli/default.nix:6]:
```suggestion
{ lib
, stdenv
, rustPlatform
, fetchCrate
, pkg-config
, openssl
, darwin
```

**Review by wegank — APPROVED** [review]

**Pull request merged**